src/core/client.js

```diff
@@ -211,8 +212,8 @@ export class Client {
     }
 
     if (timeout < 0) {
-      Log.info(`Default timeout is less than 0. Setting default timeout to ${defaultTimeout}ms.`);
-      timeout = defaultTimeout;
+      Log.info(`Default timeout is less than 0. Setting default timeout to ${this.defaultTimeout}ms.`);
```

Comment:
Should this use `DEFAULT_TIMEOUT`?

Reply:
I don't think so. The instance value is set in the constructor and that features `DEFAULT_TIMEOUT` as the default or something.

Reply:
This is correct but I was thinking about a different use case. For example, I start out by setting the `defaultTimeout` for the `client` instance to `-1` which results in the value being set to `DEFAULT_TIMEOUT`.

```js
const client = new Client({ defaultTimeout: -1 });
// client.defaultTimeout === DEFAULT_TIMEOUT (60000)
```

But later on lets say I change the `defaultTimeout` to 30s.

```js
client.defaultTimeout = 30000
```

And then somewhere else I set the default timeout to -1

```js
client.defaultTimeout = -1;
```

This will set the `defaultTimeout` to the previous value of 30s. Is this the intended result we want or should it set it to `DEFAULT_TIMEOUT`? I think it makes sense to use the previous value and the way it works now is the intended result but I wanted to double check.

I am aware I wrote this code but now I am getting the chance to review it again 😄.

Reply:
Um... if you pass an invalid parameter, this should result in a runtime exception. I'm not a fan of this sort of fault tolerance :) This is my opinion. But this is out of scope anyways. It can be examined separately from the redesign.

This is just fixing an error which would have resulted in a `ReferenceError`. It was pointed out to me by the linter.

Reply:
Ok I will add a JIRA ticket to capture this somewhere else since this is out of scope.

Reply:
I believe [MLIBZ-2251](https://kinvey.atlassian.net/browse/MLIBZ-2251) is the ticket for this.